src/coreclr/vm/peassembly.cpp

```diff
@@ -320,12 +324,40 @@ void PEAssembly::ConvertMDInternalToReadWrite()
     }
     CONTRACTL_END;
 
-    IMDInternalImport *pOld;            // Old (current RO) value of internal import.
-    IMDInternalImport *pNew = NULL;     // New (RW) value of internal import.
-
-    // Take a local copy of *ppImport.  This may be a pointer to an RO
+    // Take a local copy of the assembly's MD import.  This may be a pointer to an RO
     //  or to an RW MDInternalXX.
-    pOld = m_pMDImport;
+    IMDInternalImport *pNew = NULL;                                 // New (RW) value of internal import.
+    IMDInternalImport *pOld = m_pMDImport;                          // Old (current RO) value of internal import.
+    IMDInternalImport *pConvertedImport = m_pConvertedMDImport;     // Pointer to a RW internal metadata that might not be actively in use.
+
+
+    if (pConvertedImport != NULL)
```

Comment:
Can this be moved to the end of the method and merged with similar code that's there already?

I think the code would be much easier to understand if the method flow is like this:
 ```
// Step 1: Create RW MD import if it was not created yet
if (m_pConvertedMDImport == NULL)
{
    create and initialize m_pConvertedMDImport
}

// Step 2: Swap the RW import if the scope is being opened for writing and the RW import was not swapped yet
if (openForWriting && m_pConvertedMDImport != m_pMDImport)
{
...
}
```